### PR TITLE
Fix product node issues on Windows

### PR DESCRIPTION
**Pull request by valgur**
Fixes the test failures that cropped up in #438.

- Mark downloaded data as 'binary' in git to prevent the inserted line break characters on Windows from messing up cassette playback. The downloaded files used in test are now stored in `fixtures/vcr_cassettes/data` to make them easier to distinguish.
- Make `/` the path separator for within-product files irregardless of the OS path separator. I think that's more convenient and expected behavior for the user. Do you agree @avalentino?
- Some minor code tidying.

### CODE REVIEW

File: sentinelsat/products.py [hunk 1]
```diff
@@ -86,28 +86,25 @@ def nodefilter(node_info: dict) -> bool:
...
         elif urltype is None:
-            data["urltype"] = ""
-        else:
-            data["urltype"] = urltype
-        return "{api_url}odata/v1/Products('{id}')/Nodes('{title}.SAFE')/{path}{urltype}".format(
-            **data
-        )
+            urltype = ""
+        # else: pass urltype as is
+        return f"{self.api_url}odata/v1/Products('{id}')/Nodes('{title}.SAFE')/{path}{urltype}"
```
Comment (avalentino) [inline sentinelsat/products.py:101]: +1, but you need to drop support Python 3.5 to use it. In ``setap.py`` 3.5 is still listed.

**Review by avalentino — APPROVED** [review]
LGTM

**Comment by avalentino** [comment]
> Make / the path separator for within-product files irregardless of the OS path separator. I think that's more convenient and expected behavior for the user. Do you agree @avalentino?

@valgur absolutely yes :)

**Review by j08lue — APPROVED** [review]
Nice fixes and improvements.

**Pull request merged**